src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js

```diff
@@ -119,8 +129,41 @@ export class DeviceDetails extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    if ((this.props.device || {}).id !== nextProps.device.id) {
-      const deviceId = (nextProps.device || {}).id;
+    const {
+      deviceModuleStatus,
+      isDeviceModuleStatusPending,
+      deviceModuleStatusError,
+      moduleStatus,
+      resetPendingAndError,
+      device,
+      fetchModules
+    } = nextProps;
+
+    /*
+      deviceModuleStatus is a prop fetched by making fetchModules() API call through deviceDetails.container on demand.
+      moduleStatus is a prop sent from deploymentDetailsGrid which it already has in rowData.
+      Both deviceModuleStatus and moduleStatus have the same content,
+        but come from different sources based on the page that opens this flyout.
+      Depending on which one is available, currentModuleStatus is set in component state.
+    */
+
+    // set deviceModuleStatus in state if moduleStatus doesn't exist and devicesReducer successfully received the API response
+    if (!moduleStatus && !isDeviceModuleStatusPending && !deviceModuleStatusError) {
+      console.log(deviceModuleStatus);
+      this.setState({ currentModuleStatus: deviceModuleStatus });
+    }
+
+    // Reset state if the device changes
+    if ((this.props.device || {}).id !== device.id) {
+      this.setState(this.baseState);
+      resetPendingAndError();
+      // If moduleStatus exist in props, set it in state, otherwise make an API call to get deviceModuleStatus.
+      if (moduleStatus) {
+        this.setState({ currentModuleStatus: moduleStatus });
```

Comment:
Let's try to avoid making so many set state calls back to back. Better to identify all the changes that need to be made and them make them all in a single setState call.

Reply:
done